hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/DecodingFacade.java

```diff
@@ -1,33 +1,255 @@
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var senderID = convertAddressBytesToAccountID((byte[]) decodedArguments.get(1));
+		final var recipientID = convertAddressBytesToAccountID((byte[]) decodedArguments.get(2));
+		final var serialNumber = BigInteger.valueOf((long) decodedArguments.get(3));
+	}
+
+	public static void decodeMintToken(final Bytes input) {
+		final Tuple decodedArguments =
+				MINT_TOKEN_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var amount = BigInteger.valueOf((long) decodedArguments.get(1));
+		final var metadata = String.valueOf(decodedArguments.get(2));
+	}
+
+	public static void decodeBurnToken(final Bytes input) {
+		final Tuple decodedArguments =
+				BURN_TOKEN_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var amount = BigInteger.valueOf((long) decodedArguments.get(1));
+		final var serialNumbers = convertLongArrayToBigIntegerArray((long[]) decodedArguments.get(2));
+	}
+
+	public static void decodeAAssociateTokens(final Bytes input) {
+		final Tuple decodedArguments =
+				ASSOCIATE_TOKENS_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var accountID = convertAddressBytesToAccountID((byte[]) decodedArguments.get(0));
+		final var tokenIDs = decodeTokenIDsFromBytesArray((byte[][]) decodedArguments.get(1));
+	}
+
+	public static void decodeDissociateTokens(final Bytes input) {
+		final Tuple decodedArguments =
+				DISSOCIATE_TOKENS_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var accountID = convertAddressBytesToAccountID((byte[]) decodedArguments.get(0));
+		final var tokenIDs = decodeTokenIDsFromBytesArray((byte[][]) decodedArguments.get(1));
+	}
+
 	/* NOT A REAL IMPLEMENTATION */
 	public SyntheticTxnFactory.NftMint decodeMint(final Bytes input) {
-		final var tokenAddress = Address.wrap(
-				input.slice(ADDRESS_SKIP_BYTES_LENGTH + FUNCTION_SELECTOR_BYTES_LENGTH, ADDRESS_BYTES_LENGTH)
-		);
+		final Tuple decodedArguments =
+				MINT_TOKEN_FUNCTION.getOutputs().decode(input.slice(FUNCTION_SELECTOR_BYTES_LENGTH,
+						input.size() - FUNCTION_SELECTOR_BYTES_LENGTH).toArray());
+
+		final var tokenID = convertAddressBytesToTokenID((byte[]) decodedArguments.get(0));
+		final var amount = BigInteger.valueOf((long) decodedArguments.get(1));
```

Comment:
I think we should add `amount` field to the `SyntheticTxnFactory.NftMint` and rename it to something like `MintInput` since we don't have a way to know before decoding whether the input is for an Nft or a Non-fungible token. Also, in `computeMintToken` we shouldn't hardcode the `amount` to zero, we should instead use the amount from the new `MintInput` type.